File: tic_tac_toe-master/app/src/main/java/com/dhruvil/tictactoc/MainActivity.java
package com.dhruvil.tictactoc;

import androidx.appcompat.app.AppCompatActivity;

import android.graphics.drawable.Drawable;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    ImageView b1, b2, b3, b4, b5, b6, b7, b8, b9;
    Button reset;
    int ope = 0;
    int i1, i2, i3, i4, i5, i6, i7, i8, i9;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        b1 = findViewById(R.id.b1);
        b2 = findViewById(R.id.b2);
        b3 = findViewById(R.id.b3);
        b4 = findViewById(R.id.b4);
        b5 = findViewById(R.id.b5);
        b6 = findViewById(R.id.b6);
        b7 = findViewById(R.id.b7);
        b8 = findViewById(R.id.b8);
        b9 = findViewById(R.id.b9);
        reset = findViewById(R.id.reset);

        i1 = i2 = i3 = i4 = i5 = i6 = i7 = i8 = i9 = 0;

        b1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                if (i1 == 0){
                    if (ope % 2 == 0) {
                        b1.setImageResource(R.drawable.cross);
                    } else {
                        b1.setImageResource(R.drawable.circle);
                    }
                    ope++;
                }
                i1++;
            }
        });
        b2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                if (i2 == 0){
                    if (ope % 2 == 0) {
                        b2.setImageResource(R.drawable.cross);
                    } else {
                        b2.setImageResource(R.drawable.circle);
                    }
                    ope++;
                }
                i2++;
            }
        });
        b3.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                if (i3 == 0){
                    if (ope % 2 == 0) {
                        b3.setImageResource(R.drawable.cross);
                    } else {
                        b3.setImageResource(R.drawable.circle);
                    }
                    ope++;
                }
                i3++;
            }
        });
        b4.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                if (i4 == 0){
                    if (ope % 2 == 0) {
                        b4.setImageResource(R.drawable.cross);
                    } else {
                        b4.setImageResource(R.drawable.circle);
                    }
                    ope++;
                }
                i4++;
            }
        });
        b5.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                if (i5 == 0){
                    if (ope % 2 == 0) {
                        b5.setImageResource(R.drawable.cross);
                    } else {
                        b5.setImageResource(R.drawable.circle);
                    }
                    ope++;
                }
                i5++;
            }
        });
        b6.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                if (i6 == 0){
                    if (ope % 2 == 0) {
                        b6.setImageResource(R.drawable.cross);
                    } else {
                        b6.setImageResource(R.drawable.circle);
                    }
                    ope++;
                }
                i6++;
            }
        });
        b7.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                if (i7 == 0){
                    if (ope % 2 == 0) {
                        b7.setImageResource(R.drawable.cross);
                    } else {
                        b7.setImageResource(R.drawable.circle);
                    }
                    ope++;
                }
                i7++;
            }
        });
        b8.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                if (i8 == 0){
                    if (ope % 2 == 0) {
                        b8.setImageResource(R.drawable.cross);
                    } else {
                        b8.setImageResource(R.drawable.circle);
                    }
                    ope++;
                }
                i8++;
            }
        });
        b9.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                if (i9 == 0){
                    if (ope % 2 == 0) {
                        b9.setImageResource(R.drawable.cross);
                    } else {
                        b9.setImageResource(R.drawable.circle);
                    }
                    ope++;
                }
                i9++;
            }
        });

    }
}

File: StudentDataBase/app/src/main/java/com/example/studentdatabase/screen/homescreen/view/MainActivity.kt
package com.example.studentdatabase.screen.homescreen.view

import android.app.Dialog
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.TextView
import androidx.recyclerview.widget.LinearLayoutManager
import com.example.studentdatabase.R
import com.example.studentdatabase.databinding.ActivityMainBinding
import com.example.studentdatabase.screen.homescreen.controllarr.HomeAdapter
import com.example.studentdatabase.screen.homescreen.model.ModelData
import com.example.studentdatabase.utils.Dbhelper

class MainActivity : AppCompatActivity() {

    lateinit var binding: ActivityMainBinding

    var list = arrayListOf<ModelData>()

    lateinit var db: Dbhelper

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        db = Dbhelper(this)

        list = db.readData()
        setUpRv(list)

        binding.filter.setOnClickListener {
            var dialog = Dialog(this)
            dialog.setContentView(R.layout.choise_item)
            dialog.show()

            var std1 = dialog.findViewById<TextView>(R.id.std1)
            var std2 = dialog.findViewById<TextView>(R.id.std2)
            var std3 = dialog.findViewById<TextView>(R.id.std3)
            var std4 = dialog.findViewById<TextView>(R.id.std4)
            var std5 = dialog.findViewById<TextView>(R.id.std5)
            var std6 = dialog.findViewById<TextView>(R.id.std6)
            var std7 = dialog.findViewById<TextView>(R.id.std7)
            var std8 = dialog.findViewById<TextView>(R.id.std8)
            var std9 = dialog.findViewById<TextView>(R.id.std9)
            var std10 = dialog.findViewById<TextView>(R.id.std10)
            var std11 = dialog.findViewById<TextView>(R.id.std11)
            var std12 = dialog.findViewById<TextView>(R.id.std12)
            var alldata = dialog.findViewById<TextView>(R.id.alldata)

            alldata.setOnClickListener{

                list = db.readData()
                setUpRv(list)
            }

            std1.setOnClickListener{

                var std =1
                list = db.filterData(std)
                setUpRv(list)
            }

            std2.setOnClickListener{

                var std =2
                list = db.filterData(std)
                setUpRv(list)
            }

            std3.setOnClickListener{

                var std =3
                list = db.filterData(std)
                setUpRv(list)
            }

            std4.setOnClickListener{

                var std =4
                list = db.filterData(std)
                setUpRv(list)
            }

            std5.setOnClickListener{

                var std =5
                list = db.filterData(std)
                setUpRv(list)
            }

            std6.setOnClickListener{

                var std =6
                list = db.filterData(std)
                setUpRv(list)
            }

            std7.setOnClickListener{

                var std =7
                list = db.filterData(std)
                setUpRv(list)
            }

            std8.setOnClickListener{

                var std =8
                list = db.filterData(std)
                setUpRv(list)
            }

            std9.setOnClickListener{

                var std =9
                list = db.filterData(std)
                setUpRv(list)
            }

            std10.setOnClickListener{

                var std =10
                list = db.filterData(std)
                setUpRv(list)
            }

            std11.setOnClickListener{

                var std =11
                list = db.filterData(std)
                setUpRv(list)
            }

            std12.setOnClickListener{

                var std =12
                list = db.filterData(std)
                setUpRv(list)
            }

        }



    }

    fun setUpRv(list: ArrayList<ModelData>) {

        var adapter = HomeAdapter(this,list)
        var lm = LinearLayoutManager(this)
        binding.rvview.layoutManager = lm
        binding.rvview.adapter = adapter

    }

}